src/plot_api/helpers.js

```diff
@@ -484,3 +484,19 @@ exports.manageArrayContainers = function(np, newVal, undoit) {
         np.set(newVal);
     }
 };
+
+var ATTR_TAIL_RE = /(\.[^\[\]\.]+|\[[^\[\]\.]+\])$/;
```

Comment:
good call - also added some tests for `hasParent` -> https://github.com/plotly/plotly.js/pull/1403/commits/5fb462fa6b8c256e5fc179c028d59c10ce5c7b6d